docker/docker.go

```diff
@@ -85,6 +83,12 @@ func main() {
 
 	setDefaultConfFlag(flTrustKey, defaultTrustKeyFile)
 
+	// Regardless of whether the user sets it to true or false, if they
+	// specify --tlsverify at all then we need to turn on tls
+	if *flTlsVerify {
+		*flTls = true
+	}
```

Comment:
I think this is why you have a broken CI test on janky.. currently **TestRunTLSverify** assumes that specifying `--tls-verify` without enabling tls will cause a `docker run` failure, but it isn't anymore.